src/main/java/com/ericsson/gerrit/plugins/eiffel/storage/EventStorage.java

```diff
@@ -0,0 +1,98 @@
+/*
+   Copyright 2019 Ericsson AB.
+   For a full list of individual contributors, please see the commit history.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.ericsson.gerrit.plugins.eiffel.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.net.ConnectException;
+import java.sql.SQLException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.ericsson.gerrit.plugins.eiffel.exceptions.NoSuchElementException;
+import com.ericsson.gerrit.plugins.eiffel.handlers.DatabaseHandler;
+import com.ericsson.gerrit.plugins.eiffel.handlers.Table;
+
+public abstract class EventStorage {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(EventStorage.class);
+    protected File pluginDir;
+
+    public EventStorage(File pluginDir) {
+        this.pluginDir = pluginDir;
+    }
+
+    public abstract String getEventId(String project, String searchCriteria) throws NoSuchElementException, ConnectException, FileNotFoundException;
+
+    public abstract void saveEventId(String eiffelEventId, EiffelEvent eiffelEvent)
+            throws NoSuchElementException, SQLException, ConnectException;
+
+    protected String getLastSavedEiffelEvent(String project, String searchCriteria,
+            Table tableName)
+            throws NoSuchElementException, FileNotFoundException, ConnectException {
+        try {
+            return getEventId(project, searchCriteria, tableName);
+        } catch (NoSuchElementException e) {
+            throw new NoSuchElementException(
+                    "Database did not return any value for this query\n" + "Exception Message:" + e.getMessage());
+        }
```

Comment:
What is the benefit of catching and rethrowing the same exception? I don't feel this text adds any information.

Reply:
No value here, we removed it ;)